avocado/core/test.py

```diff
@@ -383,6 +384,11 @@ def _start_logging(self):
                                                        formatter,
                                                        self._ssh_logfile)
 
+        if isinstance(sys.stdout, output.LoggingFile):
+            sys.stdout.add_logger(logging.getLogger("avocado.test.stdout"))
+        if isinstance(sys.stderr, output.LoggingFile):
+            sys.stderr.add_logger(logging.getLogger("avocado.test.stderr"))
+
```

Comment:
Thanks, I'm looking at them. I re-triggered them to have another set of "failure" and it identified yet another timing-sensitive test: https://travis-ci.org/avocado-framework/avocado/jobs/178628886#L1196

that selftest requires the test to be in one of the `try/catch` parts of the `Test._run_avocado` 1s, which apparently it does not (from time to time). I reproduced it locally using `sleep` with the same results:

```diff
diff --git a/avocado/core/test.py b/avocado/core/test.py
index a1c7c38..ce36f0a 100644
--- a/avocado/core/test.py
+++ b/avocado/core/test.py
@@ -442,6 +442,7 @@ class Test(unittest.TestCase):
             stacktrace.log_exc_info(sys.exc_info(), logger='avocado.test')
             details = sys.exc_info()[1]
             raise exceptions.TestSetupFail(details)
+        time.sleep(1.2)
         try:
             testMethod()
         except exceptions.TestSkipError as details:
```

does not produce the traceback

```diff
diff --git a/avocado/core/test.py b/avocado/core/test.py
index a1c7c38..653df1b 100644
--- a/avocado/core/test.py
+++ b/avocado/core/test.py
@@ -432,6 +432,7 @@ class Test(unittest.TestCase):
         stderr_check_exception = None
         try:
             self.setUp()
+            time.sleep(1.2)
         except exceptions.TestSkipError as details:
             stacktrace.log_exc_info(sys.exc_info(), logger='avocado.test')
             raise exceptions.TestSkipError(details)
```

does produces the traceback. It's actually logical. If we want to depend on this we have to do the `try/catch` in `_run_test`, but I think we should rather just remove the test and describe in documentation that the traceback is only shown when test is interrupted in the test code and not during pre/post process.